hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/PipelineSyncTask.java

```diff
@@ -66,21 +71,34 @@ public PipelineSyncTask(ReconPipelineManager pipelineManager,
   @Override
   protected synchronized void run() {
     try {
+      lock.writeLock().lock();
```

Comment:
writelock is taken outside while loop, external trigger can never happen and will cause deadlock

Reply:
Thanks @sumitagrawl for review, its fixed. pls re-review.